modyn/supervisor/internal/triggers/performancetrigger.py

```diff
@@ -0,0 +1,300 @@
+
+            else:
+                # evaluate the decision policies
+                triggered = False
+                for policy_name, policy in self.decision_policies.items():
+                    policy_decisions[policy_name] = policy.evaluate_decision(
+                        update_interval=self.config.detection_interval_data_points,
+                        evaluation_scores=evaluation_scores,
+                        data_density=self.data_density,
+                        performance_tracker=self.performance_tracker,
+                        mode=self.config.mode,
+                        method=self.config.forecasting_method,
+                    )
+                    if policy_decisions[policy_name]:
+                        triggered |= True
+
+            if triggered:
+                for policy in self.decision_policies.values():
+                    policy.inform_trigger()  # resets the internal state (e.g. misclassification counters)
+
+            trigger_idx = processing_head_in_batch - 1
+
+            drift_eval_log = PerformanceTriggerEvalLog(
+                triggered=triggered,
+                trigger_index=trigger_idx,
+                evaluation_interval=(
+                    next_detection_interval[0][1],
+                    next_detection_interval[-1][1],
+                ),
+                num_samples=num_samples,
+                num_misclassifications=num_misclassifications,
+                evaluation_scores=evaluation_scores,
+                policy_decisions=policy_decisions,
+            )
+            if log:
+                log.evaluations.append(drift_eval_log)
+
+            self._last_detection_interval = next_detection_interval
+            if triggered:
+                yield trigger_idx
+
+    def inform_new_model(self, most_recent_model_id: int) -> None:
+        self.most_recent_model_id = most_recent_model_id
+        self.model_refresh_needed = True
+
+        # Perform an evaluation of the NEW model on the last evaluation interval, we will derive expected performance
+        # forecasts from these evaluations.
+        num_samples, num_misclassifications, evaluation_scores = self._run_evaluation(
+            interval_data=self._last_detection_interval
+        )
+
+        self.performance_tracker.inform_trigger(
+            num_samples=num_samples,
+            num_misclassifications=num_misclassifications,
+            evaluation_scores=evaluation_scores,
+        )
+
+    # ---------------------------------------------------------------------------------------------------------------- #
+    #                                                     Internal                                                     #
+    # ---------------------------------------------------------------------------------------------------------------- #
+
+    def _run_evaluation(self, interval_data: list[tuple[int, int]]) -> tuple[int, int, dict[str, float]]:
```

Comment:
This function seems to be missing a test

Reply:
didn't add a tests as the equivalent method of datadrifttrigger also didn't have one. I'll add one for this here :)

Reply:
if it's much overhead, then i guess its fine if you see it working properly. we didnt have codecov at the time so probably I just often missed missing tests :D

Reply:
Okay, I won't add tests for this function. This would result in more mocking than actual testing.